ports/nrf/common-hal/alarm/pin/PinAlarm.c

```diff
@@ -0,0 +1,241 @@
+/*
+ * This file is part of the MicroPython project, http://micropython.org/
+ *
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2020 Dan Halbert for Adafruit Industries
+ * Copyright (c) 2020 Scott Shawcroft for Adafruit Industries
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+#include "py/runtime.h"
+#include <stdio.h>
+#include <assert.h>
+
+#include "shared-bindings/alarm/pin/PinAlarm.h"
+#include "shared-bindings/microcontroller/__init__.h"
+#include "shared-bindings/microcontroller/Pin.h"
+
+#include "nrfx.h"
+#include "nrf_gpio.h"
+#include "nrfx_gpiote.h"
+#include "nrf_soc.h"
+#include <string.h>
+
+#include "supervisor/serial.h"  // dbg_print
+
+#define WPIN_UNUSED 0xFF
+volatile char _pinhandler_gpiote_count;
+volatile nrfx_gpiote_pin_t _pinhandler_ev_pin;
+#define MYGPIOTE_EV_PIN_UNDEF 0xFF
+static bool pins_configured = false;
+
+extern uint32_t reset_reason_saved;
+extern void dbg_dump_GPIOregs(void);
+
+void common_hal_alarm_pin_pinalarm_construct(alarm_pin_pinalarm_obj_t *self, mcu_pin_obj_t *pin, bool value, bool edge, bool pull) {
+    if (edge) {
+        mp_raise_ValueError(translate("Cannot wake on pin edge. Only level."));
```

Comment:
`"Cannot wake on pin edge. Only level."`  -> `"Can wake only on pin level, not edge"`